src/utils/disk_utils.hpp

```diff
@@ -25,4 +25,13 @@ inline std::optional<std::string> GetOldDiskKeyOrNull(storage::Delta *head) {
   return std::nullopt;
 }
 
+inline uint64_t GetEarliestTimestamp(storage::Delta *head) {
+  uint64_t ts = head->timestamp->load(std::memory_order_acquire);
+  while (head->next != nullptr) {
+    head = head->next;
+    ts = std::min(ts, head->timestamp->load(std::memory_order_acquire));
```

Comment:
Why min, shouldn't the next delta always be older

Reply:
We are looking for the earliest timestamp.
But probably that's wrong and we should look for the latest.
Because if the object was changed in transaction we should work with the latest timestamp in mvcc functions.

Reply:
Not sure what exactly you are trying to do, but know that there is a possibility that the oldest object may be scheduled for garbage collection. In any case, the memory won't get released if the gc can't get an exclusive lock.
Can this impact you in any way?

Reply:
there is no gc in disk storage, as long as the transaction exists, objects are alive

Reply:
We found a bug: https://github.com/memgraph/memgraph/issues/1195
Should be addressed separately